### PR TITLE
Fix incorrect warning when tutorial content references an article

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://87552238

## Summary

This fixed an incorrect warning when a tutorial references an article that's automatically curated. 

Because the article isn't registered yet when the tutorial resolves its references, this always raises a warning about the unresolved reference, but during rendering the reference may still resolve.

## Dependencies

n/a

## Testing

In a documentation catalog with som tutorial content and wit symbol graph files from a single module;
- Add a new article without curating it
- Reference that article from some tutorial content.
There shouldn't be a warning about the article reference failing to resolve (and when viewing that tutorial page.

> Note that these changes only affect the warning. Even without these changes the reference may still resolve during rendering and display the expected link on the rendered tutorial page.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

File: Sources/SwiftDocC/Infrastructure/DocumentationContext.swift [hunk 1]
```diff
@@ -1761,6 +1761,13 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         
         try shouldContinueRegistration()
         
+        // Articles that will be automatically curated can be resolved but they need to be pre registered before resolving links.
+        let rootNodeForAutomaticCuration = rootModules.count == 1 ? rootModules.first.flatMap(topicGraph.nodeWithReference) : nil
+        if rootNodeForAutomaticCuration != nil {
+            registerArticles(otherArticles, in: bundle)
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Infrastructure/DocumentationContext.swift:1767]: I'm aware that the order that task are performed in this method does matter but I believe that registering articles earlier is a safe change to make.

File: Tests/SwiftDocCTests/Infrastructure/DocumentationContextTests.swift [hunk 2]
```diff
@@ -1097,6 +1097,81 @@ class DocumentationContextTests: XCTestCase {
         )
     }
     
+    func testCanResolveArticleFromTutorial() throws {
```
Comment (franklinsch) [inline Tests/SwiftDocCTests/Infrastructure/DocumentationContextTests.swift:1100]: Nice test!

**Review by franklinsch — APPROVED** [review]

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**